master/internal/api_agents.go

```diff
@@ -4,12 +4,18 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
 )
 
 func (a *apiServer) GetAgents(
 	_ context.Context, req *apiv1.GetAgentsRequest) (resp *apiv1.GetAgentsResponse, err error) {
-	err = a.actorRequest("/agents", req, &resp)
+	response := a.m.system.AskAt(
+		actor.Addr("resourceProviders"), sproto.GetEndpointActorName{}).Get()
```

Comment:
> Isn't this the currently implemented version?

Right now, the `ResourceProviders` actor (address `"/resourceProviders"`) forwards the message to its `resourceProvider` (`"/defaultRP"` or `"/kubernetesRP"`), which then sends back the address of a third actor (`"/agents"` or `"/pods"`). Instead, I proposed that `"/resourceProviders"` could send back `"/defaultRP"` or `"/kubernetesRP"` directly.

The point about blocking is a big one, maybe enough to outweigh the points below; let's go ahead with this as it is, but, for the record, here's what I was thinking for the alternative:
- That makes the message part of the interface of the RP rather than the address of the internal actor, which matches the general usage of the actor system better.
- It makes the locations where API requests are served a bit more consistent.

Reply:
Yeah I agree with this. As we talked offline going to just leave this as if for now since we expect logic to have to change in the near future to support multiple RPs.